src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs

```diff
@@ -143,6 +145,126 @@ public void GivenWorkloadManifestUpdateItCanCalculateUpdates()
             manifestUpdates.Should().BeEquivalentTo(expectedManifestUpdates);
         }
 
+        [Fact]
+        public void GivenWorkloadManifestRollbackItCanCalculateUpdates()
+        {
+            var testDir = _testAssetsManager.CreateTestDirectory().Path;
+            var featureBand = "6.0.100";
+            var dotnetRoot = Path.Combine(testDir, "dotnet");
+            var expectedManifestUpdates = new (ManifestId, ManifestVersion, ManifestVersion)[] {
+                (new ManifestId("test-manifest-1"), new ManifestVersion("5.0.0"), new ManifestVersion("4.0.0")),
+                (new ManifestId("test-manifest-2"), new ManifestVersion("3.0.0"), new ManifestVersion("2.0.0")) };
+
+            // Write mock manifests
+            var installedManifestDir = Path.Combine(testDir, "dotnet", "sdk-manifests", featureBand);
+            var adManifestDir = Path.Combine(testDir, ".dotnet", "sdk-advertising", featureBand);
+            Directory.CreateDirectory(installedManifestDir);
+            Directory.CreateDirectory(adManifestDir);
+            foreach ((var manifestId, var existingVersion, _) in expectedManifestUpdates)
+            {
+                Directory.CreateDirectory(Path.Combine(installedManifestDir, manifestId.ToString()));
+                File.WriteAllText(Path.Combine(installedManifestDir, manifestId.ToString(), _manifestFileName), GetManifestContent(existingVersion));
+            }
+
+            var rollbackDefContent = JsonSerializer.Serialize(new Dictionary<string, string>() { { "test-manifest-1", "4.0.0" }, { "test-manifest-2", "2.0.0" } });
+            var rollbackDefPath = Path.Combine(testDir, "testRollbackDef.txt");
+            File.WriteAllText(rollbackDefPath, rollbackDefContent);
+
+            var manifestDirs = expectedManifestUpdates.Select(manifest => manifest.Item1)
+                .Select(manifest => Path.Combine(installedManifestDir, manifest.ToString(), _manifestFileName))
+                .ToArray();
+            var workloadManifestProvider = new MockManifestProvider(manifestDirs);
+            var nugetDownloader = new MockNuGetPackageDownloader(dotnetRoot);
+            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(Array.Empty<string>()), new string[] { dotnetRoot });
+            var manifestUpdater = new WorkloadManifestUpdater(_reporter, workloadManifestProvider, workloadResolver, nugetDownloader, testDir, testDir);
+
+            var manifestUpdates = manifestUpdater.CalculateManifestRollbacks(rollbackDefPath);
+            manifestUpdates.Should().BeEquivalentTo(expectedManifestUpdates);
```

Comment:
should packs be reinstalled too?

Reply:
This test is just covering the manifest updater, so packs aren't actually being installed here at all.